app/models/metric/ci_mixin/capture.rb

```diff
@@ -177,7 +177,15 @@ def perf_capture(interval_name, start_time = nil, end_time = nil)
     _, t = Benchmark.realtime_block(:total_time) do
       Benchmark.realtime_block(:capture_state) { perf_capture_state }
 
-      counters_by_mor, counter_values_by_mor_and_ts = perf_collect_metrics(interval_name_for_capture, start_time, end_time)
+      begin
+        counters_by_mor, counter_values_by_mor_and_ts = perf_collect_metrics(interval_name_for_capture, start_time, end_time)
+      rescue => err
+        MiqEvent.raise_evm_event_queue(ems_for_capture_target, "perf_capture_failed", :event_details => err.message)
```

Comment:
@agrare so you want to call [authentication_check](https://github.com/ManageIQ/manageiq/blob/bc7ad9be6dbc71280cc12efc909532f95bfce495/app/models/mixins/authentication_mixin.rb#L326) on `ems_for_capture_target` instead of raising an event here?

If I do that
* I would still have to tune kubernetes `perf_collect_metrics` to raise the error instead of rescueing it
* I guess it would need some kind of throttling, as I dont want to run that on every failed `perf_collect_metrics`

Reply:
Hm great point on the rate limiting, normally in a per-ems worker we wouldn't dequeue anything more if the auth was invalid...but this isn't a per-ems-worker.